### ISSUE TITLE
no-plusplus: Add option to allow use of the operators when not in a BinaryExpression, MemberExpression, or LogicalExpression

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint#9966]
**What rule do you want to change?**

no-plusplus

**Does this change cause the rule to produce more or fewer warnings?**

Fewer when new option is used

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option. Something like: `allowStandalone` (name open to bikeshedding).

This option will specifically allow UpdateExpressions that are not part of a larger BinaryExpression, MemberExpression, or LogicalExpression. (Maybe also if it's the right-hand side of an assignment or variable initialization? Not as sure about that.)

Basically, the goal is to try to make it so that UpdateExpressions can be used in very readable, unambiguous contexts only, while still rejecting them when they might be hard to understand.

**Please provide some example code that this change will affect:**

```js
/* eslint no-plusplus: ["error", { allowStandalone: true } */
while (i--) { console.log(arr[i]); }

for (let i = 0; i < 10; ++i) { console.log(arr[i]; }

i++;

```

**What does the rule currently do for this code?**

Warns for all three examples. (Note, if allowForLoopAfterthoughts option is used, second example will not warn.)

**What will the rule do after it's changed?**

Not warn for any of the examples.

**Comment by PinkyRabbit** [comment]
I saw coding pattern with `while` in Stoyan Stefanov book and think that
```javascript
var i = somearray.length;
while (i--) {
    // some code ...
}
```
is mutch more readable than
```javascript
var i = somearray.length;
while (i) {
    i -=1;
    // some code ...
}
```
Im very new, but I think first coding pattern more intuitive.

**Comment by platinumazure** [comment]
Closing as there does not appear to be interest in accepting this proposal at this time.
